### PR TITLE
Introduce MacOSDnsServerAddressStreamProvider which correctly detect …

### PR DESCRIPTION
**Pull request by normanmaurer**
…all nameserver configuration on MacOS

Motivation:

On MacOS it is not really good enough to check /etc/resolv.conf to determine the nameservers to use. We should retrieve the nameservers using the same way as mDNSResponser and chromium does by doing a JNI call.

Modifications:

Add MacOSDnsServerAddressStreamProvider and testcase

Result:

Use correct nameservers by default on MacOS.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: pom.xml [hunk 1]
```diff
@@ -706,6 +707,7 @@
         <version>0.13.1</version>
         <configuration>
           <parameter>
+            <ignoreMissingOldVersion>true</ignoreMissingOldVersion>
```
Comment (normanmaurer) [inline pom.xml]: this was added as otherwise it would fail as there can be no previous version found of `resolver-dns-native-macos`.

**Comment by normanmaurer** [comment]
I will merge this one after the next release is out.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by anuraaga** [comment]
Hi @normanmaurer I happened to find this while looking into netty-dns resolution issues on Windows.

https://github.com/line/armeria/issues/2243

Do you think we may need to use a similar JNI approach on Windows too? https://github.com/chromium/chromium/blob/master/net/dns/dns_config_service_win.cc#L498

**Comment by normanmaurer** [comment]
@anuraaga good question... unfortunately I am not very into windows at all, so its hard for me to judge :/

**Comment by anuraaga** [comment]
@normanmaurer Yeah me too ;) Luckily the stream is pluggable so will experiment with some approaches and see if I find any need for something different.

**Comment by anuraaga** [comment]
@normanmaurer Did a deep investigation and we might want a native implementation for Windows too. Java has inherited from the old days a legacy way pre-Windows XP of enumerating DNS server addresses - list all the adapters and query registry for their DNS config

https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/windows/native/libnet/ResolverConfigurationImpl.c#L129

Now adays, it's much better to use GetAdapterAddresses, which by default only returns information about bound adapters

https://github.com/chromium/chromium/blob/master/net/dns/dns_config_service_win.cc#L498

https://docs.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersaddresses
https://docs.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersinfo

I very much don't want to learn how to use WIndows SDK for this sort of development ;) But can give it a stab if you're open to adding a windows native DNS module. What do you think?

**Comment by anuraaga** [comment]
Sorry for spamming this PR - went ahead and created an issue for more discussion #9796

**Pull request merged**